src/match/queue.ts

```diff
@@ -0,0 +1,48 @@
+import { randomUUID } from 'crypto';
+
+export const subscribe = () => {};
+type Item = {
+  id: string;
+  joinAt: number;
+};
+
+export class Queue {
+  queue: Item[] = [];
+  subscribers: any[] = [];
+  subscribe({
+    onJoin,
+    onLeave,
+  }: {
+    onJoin: (playerID: string) => void;
+    onLeave: (playerID: string) => void;
+  }) {
+    const id = randomUUID();
+    this.subscribers.push({ id, onJoin, onLeave });
```

Comment:
Hmm, two things here:

1) I'd prefer if we don't use a randomUUID for this. Let's just push on a lowercased username here. It'll be easier to debug and manage going forward. If you're really pedantic about using a globally unique-checked piece of data, we should use the user's uid in the mongo model. 

2) Do we need onJoin and onLeave? What's it for?

Reply:
`onJoin`:  call back function when player join the queue - eg: when a user join a queue, use onJoin to call join lobby api
`onLeave`: call back function when player leave the queue
(I add these into comment )